src/EventModels.ts

```diff
@@ -0,0 +1,150 @@
+export enum ThumbnailSupport {
+    START_END = 0,
+    START_WITH_DELAYED_END,
+    SINGLE_DELAYED,
+    SINGLE,
+    NONE,
+}
+
+export type Metadata =
+    | {
+          label: string
+          id: string
+          hasDuration: false
+          valueExpression: string
+          thumbnailSupport: ThumbnailSupport
+      }
+    | {
+          label: string
+          id: string
+          hasDuration: boolean
+          valueExpression?: undefined
```

Comment:
Very strange typing. What you wanted to achieve?

Reply:
A moment of madness